File: CS/AccessActionControl.Module.Win/MyFilterController.cs
using System;
using DevExpress.XtraBars;
using DevExpress.ExpressApp;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Actions;
using DevExpress.XtraEditors.Repository;
using DevExpress.ExpressApp.Win.Templates;
using DevExpress.ExpressApp.Win.Templates.ActionContainers;

namespace AccessActionControl.Module.Win {
    public class MyFilterController : ViewController {
        private ParametrizedAction dateFilterAction;
        public MyFilterController() {
            TargetViewType = ViewType.ListView;
            TargetObjectType = typeof(MyDomainObject);
            dateFilterAction = new ParametrizedAction(
                this, "MyDateFilter", PredefinedCategory.Search, typeof(DateTime));
            dateFilterAction.Execute += dateFilterAction_Execute;
        }

        void dateFilterAction_Execute(object sender, ParametrizedActionExecuteEventArgs e) {
            CriteriaOperator criterion = null;
            if (e.ParameterCurrentValue != null && e.ParameterCurrentValue.ToString() != string.Empty) {
                criterion = new BinaryOperator("CreatedOn",
                    Convert.ToDateTime(e.ParameterCurrentValue));
            }
            ((ListView)View).CollectionSource.Criteria[dateFilterAction.Id] = criterion;
        }
    }
    public class BarActionItemsCustomizationController : WindowController {
        public BarActionItemsCustomizationController() {
            TargetWindowType = WindowType.Main;
        }
        protected override void OnActivated() {
            base.OnActivated();
            BarActionItemsFactory.CustomizeActionControl += BarActionItemsFactory_CustomizeActionControl;
        }
        void BarActionItemsFactory_CustomizeActionControl(object sender, CustomizeActionControlEventArgs e) {
            if (e.Action.Id == "MyDateFilter") {
                BarEditItem barItem = (BarEditItem)e.ActionControl.Control;
                barItem.Width = 170;
                RepositoryItemDateEdit repositoryItem = (RepositoryItemDateEdit)barItem.Edit;
                repositoryItem.Mask.UseMaskAsDisplayFormat = true;
                repositoryItem.Mask.EditMask = "yyyy-MMM-dd";
                repositoryItem.NullText = "Enter date";
            }
        }
        protected override void OnDeactivated() {
            base.OnDeactivated();
            BarActionItemsFactory.CustomizeActionControl -= BarActionItemsFactory_CustomizeActionControl;
        }
    }
}

File: VB/AccessActionControl.Module.Win/MyFilterController.vb
Imports DevExpress.XtraBars
Imports DevExpress.ExpressApp
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.ExpressApp.Win.Templates
Imports DevExpress.ExpressApp.Win.Templates.ActionContainers

Namespace AccessActionControl.Module.Win
    Public Class MyFilterController
        Inherits ViewController

        Private dateFilterAction As ParametrizedAction
        Public Sub New()
            TargetViewType = ViewType.ListView
            TargetObjectType = GetType(MyDomainObject)
            dateFilterAction = New ParametrizedAction(Me, "MyDateFilter", PredefinedCategory.Search, GetType(Date))
            AddHandler dateFilterAction.Execute, AddressOf dateFilterAction_Execute
        End Sub

        Private Sub dateFilterAction_Execute(ByVal sender As Object, ByVal e As ParametrizedActionExecuteEventArgs)
            Dim criterion As CriteriaOperator = Nothing
            If e.ParameterCurrentValue IsNot Nothing AndAlso e.ParameterCurrentValue.ToString() <> String.Empty Then
                criterion = New BinaryOperator("CreatedOn", Convert.ToDateTime(e.ParameterCurrentValue))
            End If
            CType(View, ListView).CollectionSource.Criteria(dateFilterAction.Id) = criterion
        End Sub
    End Class
    Public Class BarActionItemsCustomizationController
        Inherits WindowController

        Public Sub New()
            TargetWindowType = WindowType.Main
        End Sub
        Protected Overrides Sub OnActivated()
            MyBase.OnActivated()
            AddHandler BarActionItemsFactory.CustomizeActionControl, AddressOf BarActionItemsFactory_CustomizeActionControl
        End Sub
        Private Sub BarActionItemsFactory_CustomizeActionControl(ByVal sender As Object, ByVal e As CustomizeActionControlEventArgs)
            If e.Action.Id = "MyDateFilter" Then
                Dim barItem As BarEditItem = CType(e.ActionControl.Control, BarEditItem)
                barItem.Width = 170
                Dim repositoryItem As RepositoryItemDateEdit = CType(barItem.Edit, RepositoryItemDateEdit)
                repositoryItem.Mask.UseMaskAsDisplayFormat = True
                repositoryItem.Mask.EditMask = "yyyy-MMM-dd"
                repositoryItem.NullText = "Enter date"
            End If
        End Sub
        Protected Overrides Sub OnDeactivated()
            MyBase.OnDeactivated()
            RemoveHandler BarActionItemsFactory.CustomizeActionControl, AddressOf BarActionItemsFactory_CustomizeActionControl
        End Sub
    End Class
End Namespace
